lib/cloud_controller/encryptor.rb

```diff
@@ -64,25 +104,42 @@ def encrypt(field_name, options={})
         end
 
         define_method "#{field_name}_with_encryption" do
-          VCAP::CloudController::Encryptor.decrypt send("#{field_name}_without_encryption"), send(salt_name)
+          VCAP::CloudController::Encryptor.decrypt send("#{field_name}_without_encryption"), send(salt_name), send(:key_label)
         end
         alias_method_chain field_name, 'encryption'
 
         define_method "#{field_name}_with_encryption=" do |value|
           send generate_salt_name
 
-          encrypted_value =
-            if value.blank?
-              nil
+          encrypted_value = nil
+
+          if value.blank?
+            encrypted_value = nil
+          else
+            if !VCAP::CloudController::Encryptor.current_encryption_key_label.nil? && send(:key_label) != VCAP::CloudController::Encryptor.current_encryption_key_label
```

Comment:
@ScarletTanager why do we have to call `send(:key_label)` instead of just calling the method directly?

Reply:
Seems like we don't.  I was just being consistent with the way the other property accessors were being called.  Changing it to `self.encryption_key_label` instead.